contracts/interfaces/IAxelarAmplifierGateway.sol

```diff
@@ -28,7 +90,41 @@ interface IAxelarAmplifierGateway is IAxelarGMPGateway {
 
     /**
      * @notice Executes a signed batch of commands created by verifiers on Axelar.
-     * @param  batch The signed batch.
+     * @param  signedBatch The signed batch.
+     */
+    function execute(SignedCommandBatch calldata signedBatch) external;
+
+    /**
+     * @notice Checks if a contract call is approved.
+     * @dev Determines whether a given contract call, identified by the commandId and payloadHash, is approved.
+     * @param messageId The unique identifier of the message.
+     * @param sourceChain The name of the source chain.
+     * @param sourceAddress The address of the sender on the source chain.
+     * @param contractAddress The address of the contract where the call will be executed.
+     * @param payloadHash The keccak256 hash of the payload data.
+     * @return True if the contract call is approved, false otherwise.
+     */
+    function isMessageApproved(
+        string calldata messageId,
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        address contractAddress,
+        bytes32 payloadHash
+    ) external view returns (bool);
+
+    /**
+     * @notice Validates and approves a contract call.
+     * @dev Validates the given contract call information and marks it as approved if valid.
+     * @param messageId The unique identifier of the message.
+     * @param sourceChain The name of the source chain.
+     * @param sourceAddress The address of the sender on the source chain.
+     * @param payloadHash The keccak256 hash of the payload data.
+     * @return True if the contract call is validated and approved, false otherwise.
      */
-    function execute(bytes calldata batch) external;
+    function validateMessage(
```

Comment:
Maybe `validateContractCallByMessageId` instead

Reply:
I like `validateMessage`

Reply:
it feels a bit weird with `callContract` on the source, but I agree it's simpler